### PR TITLE
refactor slack channel notifications use projects slack channels join table

### LINKED ISSUE #3118: Refactor to use project.slack_channel_codes
## Expected Behavior

Hangout/YouTube notification services should use `slack_channels` join table to ping slack based on a project's slack channels
## Current Behavior

At the moment, the values are hard coded in making it more difficult to maintain 
## Possible Solution

It would be nice to have this automated in the future, but for now this issue is just to transition to move away from hard coded values in the code base

### PR DESCRIPTION
**Pull request by mattwr18**
#### Issue addressed

fixes #3118

### CODE REVIEW

**Comment by tansaku** [comment]
1) should we also add code to do same for the youtube service?  YES
2) should we create a one-shot rake task to run through the ChannelList.rb file and populate the projects_slack_channels table? YES

File: lib/tasks/db.rake [hunk 1]
```diff
@@ -5,4 +5,18 @@ namespace :db do
       project.source_repositories.create(url: project[:github_url])
     end
   end
+
+  desc "add slack channel codes to project slack channels join table"
+  task add_project_slack_channels: :environment do
+    include ChannelsList
+    Project.all.each do |project|
+      if CHANNELS[project.slug.to_sym].respond_to? :each
```
Comment (tansaku) [inline lib/tasks/db.rake]: in principle we could use `Array()` here to simplify this code, but we'll only use it once, so if it's working it's almost not worth changing ...

**Comment by tansaku** [comment]
great work @mattwr18 - you said you tested this in Azure, so we could merge this and take it all the way to production perhaps?

**Comment by mattwr18** [comment]
yes, let's get it merged...

**Pull request merged**